source/shared/cpp/ObjectModel/MarkDownParser.cpp

```diff
@@ -0,0 +1,477 @@
+#include <vector>
+#include <iomanip>
+#include <iostream>
+#include "MarkDownParser.h"
+
+using namespace AdaptiveCards;
+
+MarkDownParser::MarkDownParser(const std::string &txt) : m_text(txt), m_curPos(m_text.begin())
```

Comment:
>m_text.begin() [](start = 79, length = 14)

Since you only initialize the current position here, this means that, unless I'm missing something here, there will be problems when doing something like this:

std::string firstString = "*hello, world*;
auto parser = MarkdownParser(firstString);
std::string parsedString = parser.TransformToHtml();
std::string wrongString = parser.TransformToHtml();

Additionally, since you provide a SetText operation, it implies you can reuse the same parser for multiple strings, but that wouldn't work.

Something that could help to solve this would be to remove the m_curPos member altogether, and make the private helpers that need it receive it as an argument.

Reply:
i'm going to add dirty bit and init method. when once used, next time it get called, it will be initialized.

---
In reply to: [154718912](https://github.com/Microsoft/AdaptiveCards/pull/1003#discussion_r154718912) [](ancestors = 154718912)